tests/models/clip/test_clip.py

```diff
@@ -120,3 +131,201 @@ def test_clip_vit_forward(self, context_length):
         assert_expected(
             torch.tensor(clip_vit_scores.embeddings_b.size()), torch.tensor((1, 12))
         )
+
+    def test_clip_vit_b16(self, data):
+        text, image, *_ = data
+        model = pretrained.clip_vit_b16(True)
+        model.eval()
+        with torch.no_grad():
+            actual_a_embedding, actual_b_embedding = model(image, text)
+
+        expected_a_embedding = torch.tensor([0.0576, -0.0506, -0.0258, 0.0152, -0.0310])
+        expected_b_embedding = torch.tensor([0.0489, -0.0210, -0.0402, -0.0558, 0.0322])
+
+        assert_expected(
+            actual=actual_a_embedding[0, :5],
+            expected=expected_a_embedding,
+            rtol=0,
+            atol=1e-4,
+        )
+
+        assert_expected(
+            actual=actual_b_embedding[0, :5],
+            expected=expected_b_embedding,
+            rtol=0,
+            atol=1e-4,
+        )
+
+    def test_clip_vit_b32(self, data):
+        text, image, *_ = data
+        model = pretrained.clip_vit_b32(True)
+        model.eval()
+        with torch.no_grad():
+            actual_a_embedding, actual_b_embedding = model(image, text)
+
+        expected_a_embedding = torch.tensor([0.0006, -0.0013, 0.0277, 0.0076, 0.0010])
+        expected_b_embedding = torch.tensor([-0.0129, 0.0152, -0.0166, 0.0414, 0.0070])
+
+        assert_expected(
+            actual=actual_a_embedding[0, :5],
+            expected=expected_a_embedding,
+            rtol=0,
+            atol=1e-4,
+        )
+
+        assert_expected(
+            actual=actual_b_embedding[0, :5],
+            expected=expected_b_embedding,
+            rtol=0,
+            atol=1e-4,
+        )
+
+    def test_clip_vit_l14(self, data):
+        text, image, *_ = data
+        model = pretrained.clip_vit_l14(True)
+        model.eval()
+        with torch.no_grad():
+            actual_a_embedding, actual_b_embedding = model(image, text)
+
+        expected_a_embedding = torch.tensor([-0.0066, 0.0428, 0.0369, 0.0357, -0.0001])
+        expected_b_embedding = torch.tensor(
+            [-0.0415, -0.0375, -0.0474, 0.0057, -0.0353]
+        )
+
+        assert_expected(
```

Comment:
assert on size

Reply:
Of the embedding or of all the parameters? If it's of the embedding, it seems unlikely the embedding size could change without changing the values.

Reply:
> unlikely the embedding size could change without changing the values.

don't quite follow. if you are going to assert on the mean value, its good to add a check on the shape of the output embedding (bsz x projection dim)?